src/streamlink_cli/main.py

```diff
@@ -138,19 +144,43 @@ def create_http_server(*_args, **_kwargs):
     return http
 
 
+def apply_format_vars(plugin, unformatted_str, fs_safe=False):
+    title = plugin.get_title() or DEFAULT_STREAM_METADATA["title"]
+    author = plugin.get_author() or DEFAULT_STREAM_METADATA["author"]
+    category = plugin.get_category() or DEFAULT_STREAM_METADATA["category"]
+    game = plugin.get_category() or DEFAULT_STREAM_METADATA["game"]
```

Comment:
You are now always computing the title, author and category metadata, category even twice, which was done lazily before. This will unnecessarily delay the stream start in some cases.

Reply:
Oh yes,

Reply:
> You can't read the stream metadata before the actual stream has been chosen, and this does happen quite late in output_stream when success_open is True, so passing another parameter to the various output_ methods is not possible.

I'm trying to understand why that's the case.  The `_get_streams` method for the plugin has already been called in `fetch_streams()` via `plugin.streams()`, so it seems like there should be valid available metadata here?  Testing seems to indicate that this is the case?